openmls/src/messages/external_proposals.rs

```diff
@@ -0,0 +1,77 @@
+//! External Proposal
+//!
+//! Contains the types and methods to build external proposal
+//! to add a client from a MLS group
+//! `Remove` & `ReInit are nto yet implemented`
+
+use crate::{
+    credentials::CredentialBundle,
+    error::LibraryError,
+    framing::{FramingParameters, MlsMessageOut, MlsPlaintext, Sender, WireFormat},
+    group::{
+        mls_group::errors::ProposeAddMemberError,
+        GroupEpoch, GroupId,
+    },
+    key_packages::KeyPackage,
+    messages::{Proposal, AddProposal}
+};
+use openmls_traits::OpenMlsCryptoProvider;
+
+/// External Proposal.
+/// External proposal allows parties outside a group to request changes to the latter.
+///
+/// This `enum` contains the different external proposals in its variants.
+/// Not yet implemented: `Remove` & `ReInit`
+pub enum ExternalProposal {
```

Comment:
Since some things are not implemented, we should add the `#[non_exhaustive]` attribute (ref: https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute)

```suggestion
#[non_exhaustive]
pub enum ExternalProposal {
```